### PR TITLE
[DBIO-2188, DBIO-2187] feat: genetic analyst order & update filter list dashboard

### PR DESCRIPTION
**Pull request by azmimuhammad**
### JIRA LINK 🟦
- https://blocksphere2020.atlassian.net/browse/DBIO-2188
- https://blocksphere2020.atlassian.net/browse/DBIO-2187

### Changelog/Description 🗒️
- table list GA orders & update filter list data on Dashboard

### WARNING ⚠️
-

### CODE REVIEW

File: src/views/Dashboard/GeneticAnalyst/Home/index.vue [hunk 1]
```diff
@@ -174,10 +179,12 @@ export default {
 
     async getOrdersData(keyword) {
       this.orderLists = []
+      this.isLoading = true
 
       try {
         let orders = []
         const orderData = await GAGetOrders(keyword)
+        const page = VueRouter?.history?.current
```
Comment (agustinustheo) [inline src/views/Dashboard/GeneticAnalyst/Home/index.vue:187]: You can access the router from `this.$router` why use `VueRouter`?

**Review by agustinustheo — CHANGES_REQUESTED** [review]

File: src/views/Dashboard/GeneticAnalyst/Home/index.vue [hunk 2]
```diff
@@ -204,12 +211,16 @@ export default {
             })
           }
 
-          orders.push(data)
+          if (page.name == "ga-orders" && data.status == "Done") orders.push(data)
+          else if (page.name == "ga-dashboard" && data.status != "Done") orders.push(data)
```
Comment (bukandicki) [inline src/views/Dashboard/GeneticAnalyst/Home/index.vue:215]: I'm just curious, can you explain how that logic works?

**Review by bukandicki — CHANGES_REQUESTED** [review]

File: src/views/Dashboard/GeneticAnalyst/Home/index.vue (hunk 1 above)
Comment (azmimuhammad) [reply src/views/Dashboard/GeneticAnalyst/Home/index.vue]: updated access from `this.$router`

File: src/views/Dashboard/GeneticAnalyst/Home/index.vue (hunk 2 above)
Comment (azmimuhammad) [reply src/views/Dashboard/GeneticAnalyst/Home/index.vue:214]: this page used in `ga-orders` & `ga-dashboard` but in different filter data. Orders page only show `status = Done` (first condition), while on the dashboard page showing other than `Done` (second condition)

Comment (agustinustheo) [reply src/views/Dashboard/GeneticAnalyst/Home/index.vue:214]: Hmm, I don't think adding global logic like this is good, can you please refactor this?

**Review by agustinustheo — CHANGES_REQUESTED** [review]

File: src/views/Dashboard/GeneticAnalyst/Layout.vue [hunk 3]
```diff
@@ -215,7 +215,7 @@ export default {
 
     goToDashboard() {
       this.showModalError = false
-      this.$router.push({ name: "dashboard"})
+      this.$router.push({ name: "dashboard", query: {}})
```
Comment (agustinustheo) [inline src/views/Dashboard/GeneticAnalyst/Layout.vue:218]: Why is this pushing an empty query? @azmimuhammad

**Review by agustinustheo — CHANGES_REQUESTED** [review]

File: src/views/Dashboard/GeneticAnalyst/Layout.vue (hunk 3 above)
Comment (azmimuhammad) [reply src/views/Dashboard/GeneticAnalyst/Layout.vue:218]: on this PR https://github.com/debionetwork/debio-genetic-analyst-ui/pull/13 push query if status response 4/5, if not pushing an empty query the error popup will always active

File: src/views/Dashboard/GeneticAnalyst/Home/index.vue (hunk 2 above)
Comment (azmimuhammad) [reply src/views/Dashboard/GeneticAnalyst/Home/index.vue]: updated to using global component and send params for the filter @agustinustheo , wdyt?

Comment (azmimuhammad) [reply src/views/Dashboard/GeneticAnalyst/Home/index.vue]: i also update updated the unit testing because there was an error

**Review by agustinustheo — APPROVED** [review]

**Review by bukandicki — APPROVED** [review]

**Pull request merged**